llms/mlx_lm/models/qwen.py

```diff
@@ -95,7 +97,7 @@ def __init__(self, args: ModelArgs):
             args.hidden_size, args.intermediate_size // 2, bias=not args.no_bias
         )
         self.w2 = nn.Linear(
-            args.intermediate_size // 2, args.hidden_size, bias=not args.no_bias
+            args.hidden_size, args.intermediate_size // 2, bias=not args.no_bias
```

Comment:
This is a bug I found in the original Qwen implementation, fixed it according to the HF repo's implementation. 
https://huggingface.co/Qwen/Qwen-1_8B-Chat/blob/main/modeling_qwen.py#L558-L568

Reply:
Interesting, I wonder how it was working before? Maybe we were not loading in strict mode so the weights were just getting set correctly anyway...

Reply:
Yeah, maybe the current example requires aconversation. so it make the mismatch dimension less obvious and example is using Chinese prompt and the performance degradation is not noticeable to most non-Chinese speakers.